src/TorSharp/Tools/WhichUtility.cs

```diff
@@ -1,15 +1,21 @@
-﻿using System.Diagnostics;
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
 using System.Text;
 
 namespace Knapcode.TorSharp.Tools
 {
-    internal static class WhereIsUtility
+    internal static class WhichUtility
     {
-        public static string WhereIs(string searchPattern)
+        public static string Which(TorSharpSettings settings, string searchPattern)
         {
+            if (settings.OSPlatform == TorSharpOSPlatform.Windows)
```

Comment:
Maybe split it into 2 classes?

Reply:
Windows has a `where` command that behaves similarly. A second class (maybe a `partial`) seems reasonable.